File: frontend/src/types/room.ts
import type { PlayerInRoom } from './player'

export interface CreateRoomResponse {
  room_id: string
  room_code: string
  host_token: string
  status: string
  created_at: string
}

export interface JoinRoomResponse {
  player_id: string
  session_token: string
  room_code: string
  room_id: string
  player_name: string
  player_count: number
  status: string
}

export interface RoomInfoResponse {
  id: string
  room_code: string
  status: string
  current_phase: string
  night_number: number
  day_number: number
  current_role_turn?: string | null
  phase_ends_at?: string | null
  current_audio_text?: string | null
  is_paused?: boolean
  is_game_over?: boolean
  winner?: string | null
  players: PlayerInRoom[]
  player_count: number
  created_at: string
}

export interface RoleResponse {
  code: string
  name: string
  side: string
  description: string
  night_order: number | null
  is_active: boolean
}

export interface RoleCartItem {
  role_code: string
  name: string
  quantity: number
}

export interface RoleCartResponse {
  room_code: string
  room_id: string
  cart: RoleCartItem[]
  total_roles: number
  total_players: number
  can_start: boolean
}

File: backend/app/schemas/room.py
"""Room schemas"""
from datetime import datetime
from typing import List

from pydantic import BaseModel

from app.schemas.player import PlayerInRoomResponse


class CreateRoomResponse(BaseModel):
    """Create room response schema"""
    room_id: str
    room_code: str
    host_token: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class RoomInfoResponse(BaseModel):
    """Room info response schema"""
    id: str
    room_code: str
    status: str
    current_phase: str
    night_number: int
    day_number: int
    players: List[PlayerInRoomResponse]
    player_count: int
    created_at: datetime

    class Config:
        from_attributes = True


class JoinRoomRequest(BaseModel):
    """Join room request schema"""
    name: str

    class Config:
        json_schema_extra = {
            "example": {
                "name": "Nguyễn Văn A"
            }
        }


class JoinRoomResponse(BaseModel):
    """Join room response schema"""
    player_id: str
    session_token: str
    room_code: str
    room_id: str
    player_name: str
    player_count: int
    status: str

    class Config:
        from_attributes = True